fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/ExpressionAnalyzer.java

```diff
@@ -331,6 +340,17 @@ public Expression visitUnboundFunction(UnboundFunction unboundFunction, Expressi
                 // so wrap COUNT with Nvl to ensure it's result is 0 instead of null to get the correct result
                 boundFunction = new Nvl(boundFunction, new BigIntLiteral(0));
             }
+
+            if (currentElementAtLevel == 1
+                    && PushDownToProjectionFunction.validToPushDown(boundFunction)) {
+                // Only rewrite the top level of PushDownToProjectionFunction, otherwise invalid slot will be generated
+                // currentElementAtLevel == 1 means at the top of element_at function, other levels will be ignored.
+                currentElementAtLevel = 0;
+                return visitElementAt((ElementAt) boundFunction, context);
+            }
+            if (unboundFunction.getName().equalsIgnoreCase("element_at")) {
```

Comment:
better to use boundFunction.getName() ?

Reply:
increase and decrease condition should be symmetrical？

Reply:
changed to `boundFunction instanceof ElementAt`